libentitlement.go

```diff
@@ -31,8 +32,22 @@ func NewEntitlementsManager(profile *secprofile.Profile) *EntitlementsManager {
 }
 
 // GetProfile() returns the current state of the security profile
-func (m *EntitlementsManager) GetProfile() *secprofile.Profile {
-	return m.profile
+func (m *EntitlementsManager) GetProfile() (*secprofile.Profile, error) {
+	if m.profile == nil {
+		return nil, fmt.Errorf("Entitlements Manager has np security profile.")
+	}
+
+	return m.profile, nil
+}
+
+// SetProfile() sets the entitlement manager's security profile
```

Comment:
Yes, if the container management platform needs to:

1. add some entitlements -> init the entitlement manager with a profile
2. some configuration which may depend on profile infos -> get EntMgr profile
3. add more custom entitlements based on overall configuration -> set EntMgr profile
4. finish configuring

I'm not sure that all platforms want to do the entitlement configuration at the exact same moment in an atomic operation.

In Moby for example, there are already plenty of codepaths to set lots of defaults for masked paths, mount points etc.. and we won't be able necessarily to move all those codepaths right away to custom entitlements. This is why we need some way to get the profile from the entitlement manager, add stuff in it directly in Moby/Kub and then update the ent mgr profile.